### PR TITLE
feat: update hybridsql docker image & hybridse workflow

### LINKED ISSUE #670: Integrate third-party dependencies into docker image (hybridsql image)
release docker images with built dependency for centos only, except zetasql

### PR DESCRIPTION
**Pull request by aceforeverd**
- add thirdparty pre-compiled into docker image, close #670 
- update thirdparty dependency in `third-party/CMakeLists.txt`
- update hybridse workflow, WIP #676 
   - rm hybridse release job
   - rm hybridse install and release artifacts

### CODE REVIEW

**Comment by aceforeverd** [comment]
[compilation improvement](https://trello.com/c/VcpqXsbY/3-compilation-improvement)

**Comment by aceforeverd** [comment]
[github workflow reorganization](https://trello.com/c/upawl5mF/8-github-workflow-reorganization)

File: hybridse/docker/Dockerfile [hunk 1]
```diff
@@ -27,15 +27,20 @@ RUN yum update -y && yum install -y centos-release-scl epel-release && \
     yum clean all && rm lcov-1.15-1.noarch.rpm && \
     curl -SLo apache-maven-3.6.3-bin.tar.gz https://mirrors.ocf.berkeley.edu/apache/maven/maven-3/3.6.3/binaries/apache-maven-3.6.3-bin.tar.gz && \
     tar xzf apache-maven-3.6.3-bin.tar.gz -C /usr/local --strip-components=1 && \
-    curl -SLo thirdsrc.tar.gz https://github.com/jingchen2222/hybridsql-asserts/releases/download/v0.3.1/thirdsrc-2021-05-17.tar.gz && \
-    mkdir -p /depends/thirdsrc && \
-    tar xzf thirdsrc.tar.gz -C /depends/thirdsrc --strip-components=1 && \
+    curl -SLo thirdparty.tar.gz https://github.com/4paradigm/hybridsql-asserts/releases/download/v0.4.2/thirdparty-2021-12-10-linux-gnu-x86_64-centos.tar.gz && \
+    mkdir -p /deps/usr && \
+    tar xzf thirdparty.tar.gz -C /deps/usr --strip-components=1 && \
```
Comment (aceforeverd) [inline hybridse/docker/Dockerfile:32]: missing zetasql

**Comment by aceforeverd** [comment]
cicd/build_and_cpp_ut time will increase, but still less than coverage

File: hybridse/docker/Dockerfile [hunk 2]
```diff
@@ -27,15 +27,20 @@ RUN yum update -y && yum install -y centos-release-scl epel-release && \
     yum clean all && rm lcov-1.15-1.noarch.rpm && \
```
Comment (aceforeverd) [inline hybridse/docker/Dockerfile:27]: move hybridse/docker to docker

File: third-party/CMakeLists.txt [hunk 3]
```diff
@@ -70,20 +70,17 @@ set(HYBRIDSQL_ASSERTS_HOME https://github.com/4paradigm/hybridsql-asserts)
 
 if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
   if(CMAKE_SYSTEM_PROCESSOR MATCHES "(arm64)|(ARM64)|(aarch64)|(AARCH64)")
-    set(HYBRIDSQL_ASSERTS_URL "${HYBRIDSQL_ASSERTS_HOME}/releases/download/v0.4.0/thirdparty-2021-08-03-linux-gnu-aarch64.tar.gz")
+    set(HYBRIDSQL_ASSERTS_URL "https://github.com/4paradigm/hybridsql-asserts/releases/download/v0.4.0/thirdparty-2021-08-03-linux-gnu-aarch64.tar.gz")
     set(HYBRIDSQL_ASSERTS_HASH 128e2147903c6b4f5649284326c60ba048ff369f2925f9957f2973a515c65d75)
   else()
-    set(HYBRIDSQL_ASSERTS_URL "${HYBRIDSQL_ASSERTS_HOME}/releases/download/v0.4.0/thirdparty-2021-08-03-linux-gnu-x86_64.tar.gz")
```
Comment (aceforeverd) [inline third-party/CMakeLists.txt:76]: check ubuntu

File: third-party/CMakeLists.txt [hunk 4]
```diff
@@ -70,20 +70,17 @@ set(HYBRIDSQL_ASSERTS_HOME https://github.com/4paradigm/hybridsql-asserts)
 
 if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
```
Comment (zhanghaohit) [inline third-party/CMakeLists.txt:78]: check `/deps/usr` to see whether the thirdparty deps are already downloaded.

Comment (aceforeverd) [reply third-party/CMakeLists.txt:78]: third-party is kind of a general module to download/setup thirdparty, it do not have any assume of where that thirdparty located,  and should not contains hard link with other presets. 
instead, `/deps/usr` is checked in top `Makefile`.

File: third-party/CMakeLists.txt [hunk 5]
```diff
@@ -68,22 +68,45 @@ message(STATUS "Install bundled dependencies into ${DEPS_INSTALL_DIR}")
...
+    set(LSB_RELEASE_VERSION_SHORT "${LSB_RELEASE_VERSION_SHORT}" PARENT_SCOPE)
+    set(LSB_RELEASE_CODENAME_SHORT "${LSB_RELEASE_CODENAME_SHORT}" PARENT_SCOPE)
+endfunction()
+
 if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+  get_linux_lsb_release_information()
+
   if(CMAKE_SYSTEM_PROCESSOR MATCHES "(arm64)|(ARM64)|(aarch64)|(AARCH64)")
-    set(HYBRIDSQL_ASSERTS_URL "${HYBRIDSQL_ASSERTS_HOME}/releases/download/v0.4.0/thirdparty-2021-08-03-linux-gnu-aarch64.tar.gz")
+    set(HYBRIDSQL_ASSERTS_URL "https://github.com/4paradigm/hybridsql-asserts/releases/download/v0.4.0/thirdparty-2021-08-03-linux-gnu-aarch64.tar.gz")
```
Comment (zhanghaohit) [inline third-party/CMakeLists.txt:82]: why not use `${HYBRIDSQL_ASSERTS_HOME}` here?

File: third-party/CMakeLists.txt [hunk 6]
```diff
@@ -68,22 +68,45 @@ message(STATUS "Install bundled dependencies into ${DEPS_INSTALL_DIR}")
...
+      set(HYBRIDSQL_ASSERTS_URL "https://github.com/4paradigm/hybridsql-asserts/releases/download/v0.4.2/thirdparty-2021-12-10-linux-gnu-x86_64-ubuntu.tar.gz")
+      set(HYBRIDSQL_ASSERTS_HASH 4cc0bc5f1b84e1ce6b3d95d1f0552cfabacb404e4a82651874e80859a7f59ec8)
+    else()
+      message(FATAL_ERROR "no pre-compiled thirdparty for your operation system, try compile thirdparty from source with '-DBUILD_BUNDLED=ON'")
+    endif()
   endif()
 elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-  set(HYBRIDSQL_ASSERTS_URL "${HYBRIDSQL_ASSERTS_HOME}/releases/download/v0.4.0/thirdparty-2021-08-03-darwin-x86_64.tar.gz")
-  set(HYBRIDSQL_ASSERTS_HASH 505cf5068d8d9bd2eed5496d5eda7f83f18f3f2053b064a446876998fad0dc04)
+  set(HYBRIDSQL_ASSERTS_URL "https://github.com/4paradigm/hybridsql-asserts/releases/download/v0.4.2/thirdparty-2021-12-10-darwin-i386.tar.gz")
```
Comment (zhanghaohit) [inline third-party/CMakeLists.txt:96]: i386 is a bit mis-leading...

File: third-party/CMakeLists.txt [hunk 7]
```diff
@@ -68,22 +68,45 @@ message(STATUS "Install bundled dependencies into ${DEPS_INSTALL_DIR}")
 
 set(HYBRIDSQL_ASSERTS_HOME https://github.com/4paradigm/hybridsql-asserts)
 
+function(get_linux_lsb_release_information)
+    find_program(LSB_RELEASE_EXEC lsb_release)
+    if(NOT LSB_RELEASE_EXEC)
```
Comment (zhanghaohit) [inline third-party/CMakeLists.txt]: if `lsb_release` is not found, can parse `/etc/os-release` to see the release

**Review by zhanghaohit — CHANGES_REQUESTED** [review]

File: third-party/CMakeLists.txt (hunk 5 above)
Comment (aceforeverd) [reply third-party/CMakeLists.txt:82]: artifact name is not well constructed from the version number. absolute link is easier for one to replace with a new link.
I may replace the artifact name with `thirdparty-v0.4.3-xxx.tar.gz` next release

File: third-party/CMakeLists.txt (hunk 6 above)
Comment (aceforeverd) [reply third-party/CMakeLists.txt:96]: same above

File: third-party/CMakeLists.txt (hunk 7 above)
Comment (aceforeverd) [reply third-party/CMakeLists.txt]: That file seems just work, I'm gona rm lsb_release

File: third-party/CMakeLists.txt (hunk 5 above)
Comment (aceforeverd) [reply third-party/CMakeLists.txt:82]: Just create a issue to track that https://github.com/4paradigm/hybridsql-asserts/issues/18

**Review by zhanghaohit — APPROVED** [review]
LGTM

**Review by vagetablechicken — APPROVED** [review]

**Pull request merged**